### ISSUE TITLE
Allow variadic functions to support mixtures of types and values

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#11801]
### Summary of Problem
Today functions that take a variable number of arguments cannot accept a mixture of both types and values for those arguments.  Perhaps this is not generally useful, but it means that the Reflection module's "canResolve" function cannot be applied to functions with a mix of type and value arguments.

### Steps to Reproduce

**Source Code:**

```chapel
proc foo(args ...) {
  writeln(args);
}
foo(int, "blah");
```

**Compile command:**
`chpl foo.chpl`

**Associated Future Test(s):**
TODO

### Configuration Information

- Output of `chpl --version`: 1.19.0 pre-release

**Comment by lydia-duncan** [comment]
This is likely due to #11800 (tuples cannot contain mixture of types and values)

**Comment by bradcray** [comment]
Yeah, to me this is virtually one and the same as #11800 (not that I object to there being two issues focusing on the different perspectives).

**Comment by lydia-duncan** [comment]
I can see a world where we decide to separate their implementation.  It seems unlikely, but maybe something like this would cause us to consider it?

**Comment by bradcray** [comment]
It's not just the implementation, though, it's the language definition.  That is, given:

```chapel
proc foo(x...) { ... }
```

`x` is a tuple.

**Comment by bradcray** [comment]
> Perhaps this is not generally useful

It would be useful for things like `writeln()` where users (and developers) have often wanted to be able to print types and values.
